operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> util::Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::F64(
+            GridRasterizationQueryProcessor {
+                input: self.source.query_processor()?,
+                spatial_resolution: self.spatial_resolution,
+                grid_size_mode: self.grid_size_mode,
+                tiling_specification: self.tiling_specification,
+                origin_coordinate: self.origin_coordinate,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct InitializedDensityRasterization {
+    source: Box<dyn InitializedVectorOperator>,
+    result_descriptor: RasterResultDescriptor,
+    tiling_specification: TilingSpecification,
+    radius: f64,
+    stddev: f64,
+}
+
+impl InitializedDensityRasterization {
+    fn new(
+        source: Box<dyn InitializedVectorOperator>,
+        result_descriptor: RasterResultDescriptor,
+        tiling_specification: TilingSpecification,
+        radius: f64,
+        stddev: f64,
+    ) -> Result<Self, error::Error> {
+        ensure!(
+            radius > 0.,
+            error::InvalidOperatorSpec {
+                reason: "The radius for density rasterization must be strictly positive."
+                    .to_string()
+            }
+        );
+
+        Ok(InitializedDensityRasterization {
+            source,
+            result_descriptor,
+            tiling_specification,
+            radius,
+            stddev,
+        })
+    }
+}
+
+impl InitializedRasterOperator for InitializedDensityRasterization {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> util::Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::F64(
+            DensityRasterizationQueryProcessor {
+                input: self.source.query_processor()?,
+                tiling_specification: self.tiling_specification,
+                radius: self.radius,
+                stddev: self.stddev,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct GridRasterizationQueryProcessor {
+    input: TypedVectorQueryProcessor,
+    spatial_resolution: SpatialResolution,
+    grid_size_mode: GridSizeMode,
+    tiling_specification: TilingSpecification,
+    origin_coordinate: Coordinate2D,
+}
+
+#[async_trait]
+impl RasterQueryProcessor for GridRasterizationQueryProcessor {
+    type RasterType = f64;
+
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        let grid_resolution = match self.grid_size_mode {
+            GridSizeMode::Fixed => SpatialResolution {
+                x: f64::max(self.spatial_resolution.x, query.spatial_resolution.x),
```

Comment:
I decided that it makes sense to limit the resolution to the query resolution in case the grid resolution is finer (smaller), because one would not notice the difference anyways. It avoids the computation of an unnecessarily fine grid which is then downsampled anyways.

Reply:
Sounds reasonable. Could you add this as a comment?

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/27b4d08ab72076434fed4c6edfe2b5d3b40ac068